lightning/src/ln/channel.rs

```diff
@@ -2969,6 +2969,30 @@ impl<Signer: Sign> Channel<Signer> {
 			panic!("Cannot update fee while peer is disconnected/we're awaiting a monitor update (ChannelManager should have caught this)");
 		}
 
+		// Before to propose a feerate update, check that we can actually afford the new fee compare to the holder balance.
+		if let Some(reserve) = self.counterparty_selected_channel_reserve_satoshis {
+			let num_htlcs = self.pending_inbound_htlcs.len() + self.pending_outbound_htlcs.len();
+			let total_fee = feerate_per_kw as u64 * (COMMITMENT_TX_BASE_WEIGHT + (num_htlcs as u64) * COMMITMENT_TX_WEIGHT_PER_HTLC) / 1000;
+			if self.value_to_self_msat / 1000 - total_fee < reserve {
```

Comment:
This needs to subtract the outbound pending htlc value.

Reply:
Damn, that's right!